bundles/init/src/main/java/com/adobe/cq/commerce/demandware/init/InitServlet.java

```diff
@@ -101,20 +104,21 @@ protected void doPost(SlingHttpServletRequest request, SlingHttpServletResponse
             request.getRequestPathInfo().getExtension());
     }
 
-    private void upload(SlingHttpServletRequest slingRequest, String endPoint, String assetURI) {
+    private void upload(SlingHttpServletRequest slingRequest, String endPoint, String assetURI,
+                        DemandwareClient dwClient) {
         final RequestBuilder requestBuilder = RequestBuilder.get();
         requestBuilder.setUri(assetURI);
         final Cookie token = slingRequest.getCookie("login-token");
         requestBuilder.addHeader(new BasicHeader("Cookie", token.getName() + "=" + token.getValue()));
         final HttpHost localHost = new HttpHost("localhost", 4502);
-        final HttpClient httpClient = demandwareClient.getHttpClientBuilder().build();
+        final HttpClient httpClient = clientProvider.getDefaultClient().getHttpClientBuilder().build();
```

Comment:
we can use dwClient here... If we don't have a client, this method wouldn't have been called in the first place